task1

Will-Shanks opened the pull request:
Fix #1

pep8speaks commented:
Hello @Will-Shanks! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2020-02-11 22:37:13 UTC

JadeWibbels commented on tasks/task1/Task1b_Zijun_Liu.py:
```diff
@@ -0,0 +1,60 @@
+import numpy as np
+
+
+def get_all_capitalized_word():
```
This is missing documentation.
I would also encourage splitting this function up into 3 separate functions with declarative function naming.  This will give you the ability to reuse code and make it clearer to people who might be working on or using your code in the future to know what each section does.

JadeWibbels commented on tasks/task1/Task1b_Zijun_Liu.py:
```diff
@@ -0,0 +1,60 @@
+    return result
+
+
+def get_one_column_company_name():
```
Missing documentation.
Could probably also be  split  or simplified.

JadeWibbels commented on tasks/task1/Task1b_Zijun_Liu.py:
```diff
@@ -0,0 +1,60 @@
+
+    print(one_column_company_name)
+
+
```
There are no unit tests. Setting up testing is important! Earlier the better!

JadeWibbels commented on tasks/task1/draw.py:
```diff
@@ -0,0 +1,175 @@
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG)
+    sys.exit(main())
```
Good documentation! Thanks!

Missing unit tests, and there are a couple of functions which could be split up or simplified.

JadeWibbels commented on tasks/task1/task1b.py:
```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+import sys
+
+def companyname(filepath):
```
Missing documentation. Lots of good declarative language however this could be broken up into a number of smaller more modular functions. You might find this could be simplified.  It looks like there might be elements of this file and Zijun's file that are duplicating code and effort.

Unit tests please!

Will-Shanks commented:
I just did a big clean of the branch, @JadeWibbels would you mind reviewing again?
The highlights:
- deleted stuff we're no longer planning on using
- moved all code to /src to make reuse easier for future tasks
- added an sbatch submit script to repo root dir

Known issues:
- No tests
- Needs doc pages, although doc strings/comments are fairly descriptive (IMHO)

JadeWibbels commented on src/dayToDF.py:
```diff
@@ -0,0 +1,241 @@
+    # list out all pixels that might contain a gap
+    # can exclude first/last 20 pixels as there won't be a gap that
+    #    close to the edge
+    # TODO fix magic number (20)
```
Is this to make more generalized?

JadeWibbels commented on src/dayToDF.py:
```diff
@@ -0,0 +1,241 @@
+    # append last column
+    intervals.append([start, df['xmax'].max()])
+
+    # add col column, with -1 as default
```
It might be worth breaking this up further into _get_gaps and _get_columns.

JadeWibbels commented on src/dayToDF.py:
```diff
@@ -0,0 +1,241 @@
+    return df
+
+
+def _get_lines(df):
```
This is a large function. I would encourage breaking out some of the smaller methods. Nested for loops are a good indication of where a function can be split. It will make it easier to write tests and debug each section.

Will-Shanks replied:
Are you referring to _get_cols or _get_lines?

JadeWibbels replied:
_get_lines :)

JadeWibbels commented on src/draw.py:
```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
```
This looks clean! The comments are great! Tests are trickier when dealing with plotting, but you can make some simplistic test data.

Will-Shanks replied:
Since it would be hard to write tests for this, and this code is just a tool for helping with data exploration/debugging I didn't really want to write tests for it. If you think it needs some we can talk tomorrow (Wednesday) before lecture.

JadeWibbels replied:
ah! If its for exploration and debugging dont worry about testing. Maybe just note that in the doc string :)

JadeWibbels commented on src/draw.py:
```diff
@@ -0,0 +1,175 @@
+    if len(sys.argv) >= 3:
+        try:
+            plot.setImage(sys.argv[2])
+        except FileNotFoundError:
```
Thank you for responsible use of try/except!

JadeWibbels commented on src/draw.py:
```diff
@@ -0,0 +1,175 @@
+            logging.warning("File %s not found", sys.argv[2])
+
+    # open input file
+    if len(sys.argv) < 2:
```
Do you have a catch for the wrong type of input? Is there a file type that might break this? I

JadeWibbels commented on src/getTitles.py:
```diff
@@ -0,0 +1,37 @@
+        # iter over lines in file
+        for line in dayToDF.iter_df(dayToDF.get_df(fn)):
+            # companies have a history section first
+            if line.startswith('History'):
```
This could probably be generalized too if you find that some manuals don't have this default. You could have 'History' as an arg passed or even enumerated based on year or whatever. This is minor, but worth considering if you find it has inconsistencies going forwards.

JadeWibbels commented on src/dayToDF.py:
```diff
@@ -0,0 +1,241 @@
+            print(line)
+    return 0
+
+#    code to print boxes over .tif file, helpfull for debugging
```
Is this being worked on still? Do you want to move it into master? Or remove for the purposes of this PR?

JadeWibbels commented on src/dayToDF.py:
```diff
@@ -0,0 +1,241 @@
+        but they are different
+    """
+
+    def heights(row):
```
(Feel free to use or not, I just wanted to bring this up as a suggestion.) Declarative programming is pretty hot atm, this basically just requires that you name your  functions after what they do.  For method like 'def heights' it would be 'get_row_height' (or similar). It makes it easier for others using/reading/developing to quickly identify what each piece does.

Will-Shanks commented:
Great! I'm going to squash it and merge.
